force-app/main/default/classes/RD2_CommitmentService.cls

```diff
@@ -0,0 +1,221 @@
+                'npe03__Recurring_Donation__c', String.valueOf(npe03__Recurring_Donation__c.Status__c)
+            );
+        }
+
+        List<npe03__Recurring_Donation__c> newRds = new List<npe03__Recurring_Donation__c>{rd};
+        List<npe03__Recurring_Donation__c> oldRds = new List<npe03__Recurring_Donation__c>();
+        if (rd.Id != null) {
+            oldRds.add(oldRd);
+        }
+
+        RD2_DataRegulationService regulationService = new RD2_DataRegulationService();
+        regulationService.adjust(newRds, oldRds);
+        regulationService.markRDsAsElevate(newRds);
+
+        List<ErrorRecord> errorRds = new RD2_ValidationService(newRds, oldRds)
+            .validate();
+
+        regulationService.removeElevateMarker(newRds);
+
+        if (!errorRds.isEmpty()) {
+            UTIL_AuraEnabledCommon.throwAuraHandledException(
+                errorRds[0].getFirstError()
+            );
+        }
+    }
+
+    /**
+    * @description Checks if the commitment record should be sent to Elevate
+    * @param rd Recurring Donation record
+    * @param oldRd oldRecurring Donation
+    * @param paymentMethodToken Token for Elevate requests
+    * @return Boolean
+    */
+    private Boolean shouldSendToElevate(npe03__Recurring_Donation__c rd, npe03__Recurring_Donation__c oldRd, String paymentMethodToken) {
+        if(new RD2_RecurringDonation(rd).isClosed()) {
+            return false;
+        }
+        PS_CommitmentRequest request = new PS_CommitmentRequest();
+        Boolean isElevatedFieldsChanged = request.isElevateScheduleFieldsChanged(rd, oldRd)
+            || request.isElevateCampaignChanged(rd, oldRd);
+
+
+        return rd.Id == null
+            || (isElevatedFieldsChanged && rd.CommitmentId__c != null)
+            || String.isNotBlank(paymentMethodToken);
+    }
+
+    /**
+    * @description Sends commitment request to Elevate
+    * @param commitmentId Elevate recurring commitment Id
+    * @param jsonRequestBody Payment API request in JSON format
+    * @return response Payments API response
+    */
+    private UTIL_Http.Response sendRequest(String commitmentId, String jsonRequestBody,
+        PS_Request.ElevateEndpoint endpoint) {
+        UTIL_Http.Response response;
+
+        try {
+            HttpRequest request = PS_CommitmentRequest.buildRequest(commitmentId, jsonRequestBody,
+                endpoint);
+
+            response = requestService.sendRequest(request);
+
+        } catch (Exception ex) {
+            response = requestService.buildErrorResponse(ex);
+        }
+
+        return response;
+    }
+
+    /**
+    * @description Logs an error record on an commitment error response
+    * @param rd Recurring Donation record
+    * @param response Payments API response
+    */
+    private void processResponse(npe03__Recurring_Donation__c rd, UTIL_Http.Response response) {
+        if (isCommitmentSuccess(response)) {
+            return;
+        }
+
+        Id recordId = rd.Id != null
```

Comment:
From this point forward can be abstracted out into a separate method.  It is identical to the version below.

Reply:
Updated!